arm-operationalinsights/2015-03-20/swagger/OperationalInsights.json

```diff
@@ -0,0 +1,1110 @@
+    "SearchMetadataSchema": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name."
+        },
+        "version": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The version."
+        }
+      },
+      "description": "Metadata items for saved search items."
+    },
+    "SearchMetadata": {
+      "properties": {
+        "RequestId": {
+          "x-ms-client-name": "SearchId",
+          "type": "string",
+          "description": "The request id of the search."
+        },
+        "resultType": {
+          "type": "string",
+          "description": "The search result type."
+        },
+        "total": {
+          "type": "integer",
+          "format": "int64",
+          "description": "The total search results."
+        },
+        "top": {
+          "type": "integer",
+          "format": "int64",
+          "description": "The number of top search results."
+        },
+        "id": {
+          "type": "string",
+          "description": "The id of the search results request."
+        },
+        "CoreSummaries": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CoreSummary"
+          },
+          "description": "The core summaries."
+        },
+        "Status": {
+          "type": "string",
+          "description": "The status of the search results."
+        },
+        "StartTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The start time for the search."
+        },
+        "LastUpdated": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The time of last update."
+        },
+        "ETag": {
+          "type": "string",
+          "description": "The ETag of the search results."
+        },
+        "sort": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SearchSort"
+          },
+          "description": "Gets or sets sort."
+        },
+        "requestTime": {
+          "type": "integer",
+          "format": "int64",
+          "description": "The request time."
+        },
+        "aggregatedValueField": {
+          "type": "string",
+          "description": "The aggregated value field."
+        },
+        "aggregatedGroupingFields": {
+          "type": "string",
+          "description": "The aggregated grouping fields."
+        },
+        "sum": {
+          "type": "integer",
+          "format": "int64",
+          "description": "The sum."
+        },
+        "max": {
+          "type": "integer",
+          "format": "int64",
+          "description": "The max."
+        },
+        "schema": {
+          "$ref": "#/definitions/SearchMetadataSchema",
+          "description": "The schema."
+        }
+      },
+      "description": "Metadata for search results."
+    },
+    "SavedSearch": {
+      "properties": {
+        "id": {
```

Comment:
If the id is indeed `"readOnly": true` then there is no difference between the "SavedSearchCreateOrUpdateParameters" and "SavedSearch" models. You can delete the "SavedSearchCreateOrUpdateParameters" model. The input and output for the PUT operation will be the same model. That makes it closer to the REST principles of a PUT operation (PUT either creates something new or replaces the old version of the object with a new version)